File: codechef/ECR42_2.cpp


#include <cstdio>


int n,c[2];
char s[200005];

int main()
{
    printf("%d %d\n",c[0],c[1]);
	scanf("%d %d %d",&n,c,c+1);
	scanf("%s",s+1);
	int ans=0;
	for (int i=1; i<=n; i++)
		if (s[i]=='.')
		{
			int j=i,k=(c[0]>=c[1]?0:1);
			while (j<=n&&s[j]=='.')
			{
				if (c[k]>0)
					ans++;
				c[k]--;
				k^=1;
				j++;
			}
			i=j;
		}
	printf("%d\n",ans);
	return 0;
}
/*
toggle=[1,1]
n, a, b = [int(x) for x in input().split()]
seats = input()
ans = 0
for i, v in enumerate(seats):
    if v == ".":
        j=i
        k=[1, 0][toggle[0]>=toggle[1]]
        while j < n and seats[j]==".":
            if toggle[k]>0:
                ans += 1
            toggle[k]-=1
            k^=1
            j+= 1
    i=j
print(ans)

*/

File: codechef/ECR42_2.py
'''
Examples
input
5 1 1
*...*
output
2
input
6 2 3
*...*.
output
4
input
11 3 10
.*....**.*.
output
7
input
3 2 3
***
output
0
'''

R = lambda: map(int, input().split())

n, a, b = R()
ls = tuple(map(len, input().split('*')))

s = a + b
for t in ls:
    if t % 2:
        if a > b: a -= 1
        else: b -= 1
        t -= 1
    a -= t/2; b -= t/2

print(int(s - max(0,a) - max(0,b)))



